tests/numerics/ad/test_equation_system.py

```diff
@@ -196,6 +199,77 @@ def test_variable_creation():
     )
 
 
+@pytest.mark.parametrize("variable_to_be_removed", ["var_1", "var_2", "var_3", None])
+def test_remove_variables(variable_to_be_removed):
```

Comment:
See comment to method implementation regarding modification of variable state. If the method is extended, so should the test be.